File: thor/src/jvmMain/kotlin/io/github/remmerw/thor/cobra/html/domimpl/HTMLHRElementImpl.java
package io.github.remmerw.thor.cobra.html.domimpl;

import org.w3c.dom.html.HTMLHRElement;

public class HTMLHRElementImpl extends HTMLAbstractUIElement implements HTMLHRElement {
    public HTMLHRElementImpl(final String name) {
        super(name);
    }

    public String getAlign() {
        return this.getAttribute("align");
    }

    public void setAlign(final String align) {
        this.setAttribute("align", align);
    }

    public boolean getNoShade() {
        return "noshade".equalsIgnoreCase(this.getAttribute("noshade"));
    }

    public void setNoShade(final boolean noShade) {
        this.setAttribute("noshade", noShade ? "noshade" : null);
    }

    public String getSize() {
        return this.getAttribute("size");
    }

    public void setSize(final String size) {
        this.setAttribute("size", size);
    }

    public String getWidth() {
        return this.getAttribute("width");
    }

    public void setWidth(final String width) {
        this.setAttribute("width", width);
    }
}

File: thor/src/jvmMain/kotlin/io/github/remmerw/thor/cobra/html/dom/HTMLHRElementImpl.kt
package io.github.remmerw.thor.cobra.html.dom

import org.w3c.dom.html.HTMLHRElement

class HTMLHRElementImpl(name: String) : HTMLAbstractUIElement(name), HTMLHRElement {
    override fun getAlign(): String? {
        return this.getAttribute("align")
    }

    override fun setAlign(align: String?) {
        this.setAttribute("align", align)
    }

    override fun getNoShade(): Boolean {
        return "noshade".equals(this.getAttribute("noshade"), ignoreCase = true)
    }

    override fun setNoShade(noShade: Boolean) {
        this.setAttribute("noshade", if (noShade) "noshade" else null)
    }

    override fun getSize(): String? {
        return this.getAttribute("size")
    }

    override fun setSize(size: String?) {
        this.setAttribute("size", size)
    }

    override fun getWidth(): String? {
        return this.getAttribute("width")
    }

    override fun setWidth(width: String?) {
        this.setAttribute("width", width)
    }
}
